Jenkinsfile

```diff
@@ -21,11 +21,18 @@ node ("ts-engine && heavy && java8") {
         checkout scm
         sh 'chmod +x gradlew'
     }
+
     stage('Build') {
         // Jenkins sometimes doesn't run Gradle automatically in plain console mode, so make it explicit
         sh './gradlew --console=plain clean extractConfig extractNatives distForLauncher testDist'
         archiveArtifacts 'gradlew, gradle/wrapper/*, templates/build.gradle, config/**, facades/PC/build/distributions/Terasology.zip, engine/build/resources/main/org/terasology/engine/version/versionInfo.properties, natives/**, build-logic/src/**, build-logic/*.kts'
     }
+
+    stage('Unit Tests') {
+        sh './gradlew unitTest'
+        junit testResults: '**/build/test-results/unitTest/*.xml', allowEmptyResults: true
```

Comment:
You put this stage before Publish because you want to require the tests to pass before publishing, right?

In that case, I think we want to remove `allowEmptyResults` from this one.